packages/rspeedy/lynx-bundle-rslib-config/src/lynxBundleRslibConfig.ts

```diff
@@ -0,0 +1,271 @@
+            mainThreadEntryName.push(mainThreadEntry)
+            backgroundEntryName.push(backgroundEntry)
+            // dprint-ignore
+            chain
+              .entry(mainThreadEntry)
+                .add({
+                  import: value,
+                  layer: LAYERS.MAIN_THREAD,
+                })
+                .end()
+              .entry(backgroundEntry)
+                .add({
+                  import: value,
+                  layer: LAYERS.BACKGROUND,
+                })
+                .end()
+          } else {
+            // object
+            const { layer } = value
+            if (layer === LAYERS.MAIN_THREAD) {
+              mainThreadEntryName.push(entryName)
+              // dprint-ignore
+              chain
+                .entry(entryName)
+                  .add({
+                    ...value,
+                    layer: LAYERS.MAIN_THREAD,
+                  })
+                  .end()
+            } else if (layer === LAYERS.BACKGROUND) {
+              backgroundEntryName.push(entryName)
+              // dprint-ignore
+              chain
+                .entry(entryName)
+                  .add({
+                    ...value,
+                    layer: LAYERS.BACKGROUND,
+                  })
+                  .end()
+            } else {
+              const mainThreadEntry = `${entryName}__main-thread`
+              const backgroundEntry = entryName
+              mainThreadEntryName.push(mainThreadEntry)
+              backgroundEntryName.push(backgroundEntry)
+              // dprint-ignore
+              chain
+                .entry(mainThreadEntry)
+                  .add({
+                    ...value,
+                    layer: LAYERS.MAIN_THREAD,
+                  })
+                  .end()
+                .entry(backgroundEntry)
+                  .add({
+                    ...value,
+                    layer: LAYERS.BACKGROUND,
+                  })
+                  .end()
+            }
+          }
+        }
+      })
+      // dprint-ignore
+      chain
+        .plugin(MainThreadRuntimeWrapperWebpackPlugin.name)
+        .use(MainThreadRuntimeWrapperWebpackPlugin, [{
+          test: mainThreadEntryName.map((name) => new RegExp(`${name}\\.js$`)),
+        }])
+        .end()
+        .plugin(BackgroundRuntimeWrapperWebpackPlugin.name)
+        .use(BackgroundRuntimeWrapperWebpackPlugin, [{
+          test: backgroundEntryName.map((name) => new RegExp(`${name}\\.js$`)),
+        }])
+        .end()
+    },
+  },
+}
+
+/**
+ * Get the rslib config for building Lynx external bundles.
+ *
+ * @public
+ *
+ * @example
+ *
+ * If you want to build an external bundle which use in Lynx background thread, you can use the following code:
+ *
+ * ```js
+ * // rslib.config.js
+ * import { defineExternalBundleRslibConfig, LAYERS } from '@lynx-js/lynx-bundle-rslib-config'
+ *
+ * export default defineExternalBundleRslibConfig({
```

Comment:
Got this TS error on my side:

```
The inferred type of 'default' cannot be named without a reference to '.pnpm/@rslib+core@0.15.0_@microsoft+api-extractor@7.53.1_@types+node@24.6.1__@typescript+nati_e033f1c415cf6fe54cb784797a4bdefd/node_modules/@rslib/core'. This is likely not portable. A type annotation is necessary.ts(2742)
```

<img width="966" height="435" alt="Image" src="https://github.com/user-attachments/assets/41dc3dc8-0334-4237-8f6d-ccf046ad9bf2" />

Reply:
Sorry I can't reproduce it. Could you please provide more details for me?